.github/workflows/build_release.yml

```diff
@@ -423,14 +441,39 @@ jobs:
       with:
         arch: ${{ matrix.architecture }}
 
+    - name: Restore cached ccache
+      id: ccache-cache
+      uses: actions/cache/restore@v3
+      with:
+        path: 'C:\ProgramData\chocolatey\lib\ccache\'
+        key: windows-ccache
+
     - name: update ccache
+      if: steps.ccache-cache.outputs.cache-hit != 'true'
       run: |
         choco upgrade ccache
 
+    - name: Save ccache to cache
+      if: steps.ccache-cache.outputs.cache-hit != 'true'    
+      id: check-for-ccache-save
+      uses: actions/cache/save@v3
+      with:
+        path: 'C:\ProgramData\chocolatey\lib\ccache\'
+        key: windows-ccache
+
+    - name: ccache
+      if: matrix.configuration == 'Debug'
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        key: windows-${{ matrix.configuration }}-${{ matrix.architecture }}
+        max-size: "110M"
+
     - name: ccache
+      if: matrix.configuration == 'Release'
       uses: hendrikmuhs/ccache-action@v1.2
       with:
         key: windows-${{ matrix.configuration }}-${{ matrix.architecture }}
+        max-size: "50M"
```

Comment:
That's a lot of magic numbers that might go stale at some point... Is there a way to move them out somewhere?

Reply:
Not really, I did give about a 15% buffer on them, also if they ever become too small it just means a few cache files will be ejected so there would be lets say a 95% cache hit rate rather than 100% if it goes 5% over